wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt

```diff
@@ -358,17 +367,46 @@ class MainViewModel @Inject constructor(
     fun addFavoriteEntity(entityId: String) {
         viewModelScope.launch {
             favoritesDao.addToEnd(entityId)
+            addCachedFavorite(entityId)
         }
     }
 
     fun removeFavoriteEntity(entityId: String) {
         viewModelScope.launch {
             favoritesDao.delete(entityId)
+            removeCachedFavorite(entityId)
```

Comment:
This function is only used here and doesn't do anything special, so you could call `favoriteCachesDao.delete(entityId)` here to simplify things a bit.